tests/CMakeLists.txt

```diff
@@ -12,70 +12,77 @@ foreach(src IN LISTS libsrc)
 endforeach()
 add_library(testobj OBJECT ${test_srcs} ${compatsrc})
 
-set(NEXT_TEST_PORT 10005)
+set(TEST_SRC "ln2_test.c")
+set(NEXT_TEST_PORT 10050)
+
+macro(get_test_ports PORT_COUNT PORT_DEFINITIONS)
+    if (NOT ${PORT_COUNT})
+        set(${PORT_COUNT} 1)
+    endif()
+
+    SET(PORT_INDEX 0)
+    while(PORT_INDEX LESS ${${PORT_COUNT}})
+        list(APPEND ${PORT_DEFINITIONS} "TEST_PORT_${PORT_INDEX}=${NEXT_TEST_PORT}")
+        math(EXPR PORT_INDEX "${PORT_INDEX} + 1")
+        math(EXPR NEXT_TEST_PORT "${NEXT_TEST_PORT} + 1")
+    endwhile()
+    set(NEXT_TEST_PORT ${NEXT_TEST_PORT} PARENT_SCOPE)
```

Comment:
`PARENT_SCOPE` is probably not needed, unless I am mistaken, macros share the scope of the caller, only functions do not.

Reply:
It is needed, because the macro is invoked inside a function and `NEXT_TEST_PORT` is a global variable. Can also define it as the function's variable and then it would not be needed, but this made more sense to me.

Reply:
Okay, I suppose, but seeing that in a macro is always confusing.